src/pages/search.jsx

```diff
@@ -42,27 +43,38 @@ function shouldShowResults(conditions) {
     <NoResults page={ lastPage } searchTerm={ searchTerm } />;
 }
 
-function SearchPage(props) {
-  if (
-    typeof window !== 'undefined' &&
-    !('Promise' in window)
-  ) {
-    return <NoSupport />;
+class SearchPage extends PureComponent {
+  constructor() {
+    super();
+  }
+
+  componentWillUnmount() {
+    const { resetSearch } = this.props;
+    resetSearch('');
```

Comment:
This action expects no arguments, you are creating an empty string for no reason

Reply:
Good, I'll remove the empty string.